### PR TITLE
`Development`: Add documentation about additional communication features

### PR DESCRIPTION
**Pull request by pellumbbaboci**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

missing documentation about link preview and referencing single slides.

### Description
Added documentation about link preview and referencing single slides

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: docs/user/communication.rst [hunk 1]
```diff
@@ -64,8 +64,8 @@ Users of a course can communicate in private via the Messages page. (see image b
 Conversation sidebar on the left, where users can search for other participants of the current course and start a conversation
 with them.
 
-If the recipient is browsing another conversation when they receive a new message, an envelope icon appears in their
-Conversation sidebar, next to the affiliated user who has sent the message. This way, users become aware of the new message
+If the recipient is browsing another conversation when they receive a new message, an icon with unread message count is displayed in their
```
Comment (lennart-keller) [inline docs/user/communication.rst:67]:
```suggestion
If the recipient is browsing another conversation when they receive a new message, an icon with an unread-messages counter is displayed in their
```

File: docs/user/communication.rst [hunk 2]
```diff
@@ -64,8 +64,8 @@ Users of a course can communicate in private via the Messages page. (see image b
...
+Conversation sidebar, next to the affiliated conversation that the new message was sent to. This way, users become aware of the new message
```
Comment (lennart-keller) [inline docs/user/communication.rst:68]:
```suggestion
conversation sidebar, next to the affiliated conversation that the new message was sent to. This way, users become aware of the new message
```

File: docs/user/communication.rst [hunk 3]
```diff
@@ -78,6 +78,17 @@ located on the right-hand side of the Messages Page when displayed.
...
 |messages|
 
+If the message content contains links, a preview of the link will be shown under the message. This way, users will have a good understanding
+of what that link is about. If they don't want to have the preview, by hovering over the preview, an `X` button will appear,
+and after clicking, the preview will be removed.
+
+|link-preview|
+
+If the message contains more than one link, then the preview will not have a preview image of the link to have more compact previews
+multiple links.
```
Comment (lennart-keller) [inline docs/user/communication.rst:88]:
```suggestion
If the message contains more than one link, then the preview will not have a preview image of the link to have more compact previews for multiple links.
```

File: docs/user/communication.rst [hunk 4]
```diff
@@ -182,6 +193,27 @@ Reference Lecture Attachments
...
+""""""""""""""""""""""""""""""""""
+
+Users can refer to lecture attachment units of the current course, via the dropdown menu ``Lecture`` available on the posting markdown
+editor (see image below). Here, lecture attachment units can be found when users hover over the specific lecture.
+
+Reference Lecture Unit Slides
+"""""""""""""""""""""""""""""
+
+Users can refer to lecture unit slides of the current course, via the dropdown menu ``Lecture``. Here, slides can be found when users
+hover over specific unit (see image below).
```
Comment (lennart-keller) [inline docs/user/communication.rst:206]:
```suggestion
hover over a specific unit (see image below).
```

**Review by lennart-keller — APPROVED** [review]
Doc looks good overall. Added some minor suggestions

File: docs/user/communication.rst [hunk 5]
```diff
@@ -78,6 +78,17 @@ located on the right-hand side of the Messages Page when displayed.
 
 |messages|
 
+If the message content contains links, a preview of the link will be shown under the message. This way, users will have a good understanding
+of what that link is about. If they don't want to have the preview, by hovering over the preview, an `X` button will appear,
+and after clicking, the preview will be removed.
```
Comment (nityanandaz) [inline docs/user/communication.rst:83]:
```suggestion
of what that link is about. If they prefer not to have a preview, they can hover over the preview and click the appearing `X` button.
The preview will be removed.
```
I think subjects in this sentence are a bit hard to grasp

File: docs/user/communication.rst [hunk 6]
```diff
@@ -78,6 +78,17 @@ located on the right-hand side of the Messages Page when displayed.
...
+
+|link-preview|
+
+If the message contains more than one link, then the preview will not have a preview image of the link to have more compact previews
```
Comment (nityanandaz) [inline docs/user/communication.rst:87]:
```suggestion
If the message contains more than one link, the preview will not have a preview image of the link to have more compact previews
```

File: docs/user/communication.rst [hunk 7]
```diff
@@ -182,6 +193,27 @@ Reference Lecture Attachments
 Users can refer to lectures of the current course, via the dropdown menu ``Lecture`` available on the posting markdown
 editor (see image above). Here, lecture attachments can be found in a nested structure.
 
+Reference Lecture Attachment Units
+""""""""""""""""""""""""""""""""""
+
+Users can refer to lecture attachment units of the current course, via the dropdown menu ``Lecture`` available on the posting markdown
+editor (see image below). Here, lecture attachment units can be found when users hover over the specific lecture.
```
Comment (nityanandaz) [inline docs/user/communication.rst:200]:
```suggestion
Users can refer to lecture attachment units of the current course via the dropdown menu ``Lecture`` available on the posting markdown
editor, see image below. Here, lecture attachment units can be found when users hover over the specific lecture.
```

File: docs/user/communication.rst [hunk 8]
```diff
@@ -182,6 +193,27 @@ Reference Lecture Attachments
...
+
+Reference Lecture Unit Slides
+"""""""""""""""""""""""""""""
+
+Users can refer to lecture unit slides of the current course, via the dropdown menu ``Lecture``. Here, slides can be found when users
+hover over a specific unit (see image below).
```
Comment (nityanandaz) [inline docs/user/communication.rst:206]:
```suggestion
Users can refer to lecture unit slides of the current course via the dropdown menu ``Lecture``. Here, slides can be found when users
hover over a specific unit, see image below.
```

File: docs/user/communication.rst [hunk 9]
```diff
@@ -182,6 +193,27 @@ Reference Lecture Attachments
...
+
+|slide-reference-menu|
+
+After the user references a single slide they can see as image included in the message. Additionally they can preview the slide
+in order to easily read the content by clicking the image.
```
Comment (nityanandaz) [inline docs/user/communication.rst:211]:
```suggestion
After the user references a single slide they can see an image included in the message. Additionally, they can preview the slide
in order to easily read the content by clicking the image.
```

Comment (jakubriegel) [reply docs/user/communication.rst:211]: Maybe it should be `they can see it as an image`?

**Review by jakubriegel — COMMENTED** [review]
Overall looks good to me. I agree with @nityanandaz comments

**Review by lennart-keller — DISMISSED** [review]
lgtm, thanks for including my feedback

**Review by jakubriegel — DISMISSED** [review]

**Review by nityanandaz — DISMISSED** [review]

**Review by krusche — CHANGES_REQUESTED** [review]
Some screenshots include the Development badge in the upper left corner. Please create screenshots without this and make sure those are realistic!

**Review by lennart-keller — APPROVED** [review]
Reapprove. Screenshots don't contain dev badges anymore and are realistic.

**Review by jakubriegel — APPROVED** [review]
Docs and screenshots look good

**Review by egekurt123 — APPROVED** [review]
Documentation text and screenshots look great

**Pull request merged**